weave-js/src/core/ops/primitives/date.ts

```diff
@@ -36,6 +36,96 @@ export const opDateSub = makeDateOp({
   },
 });
 
+export const opDateAddDuration = makeDateOp({
+  hidden: true,
+  name: 'date-addDuration',
+  argTypes: {date: 'date', duration: 'number'},
+  renderInfo: {
+    type: 'binary',
+    repr: '+',
+  },
+  description: `Returns the sum between a ${docType(
+    'date'
+  )} and duration in milliseconds ${docType('number')}`,
+  argDescriptions: {
+    date: `Date ${docType('date')}`,
+    duration: `Duration ${docType('number')}`,
+  },
+  returnValueDescription: `The sum between the a ${docType(
+    'date'
+  )} and duration in milliseconds ${docType('number')}`,
+  returnType: inputTypes => 'date',
+  resolver: inputs => {
+    if (
+      inputs.date == null ||
```

Comment:
not sure if this is necessary or gets auto handled

Reply:
it's supposed to be handled for you but this defensiveness here is warranted, for now.